meson.build

```diff
@@ -1713,8 +1712,22 @@ if get_option('cplayer')
                  rename: 'mpv.svg')
     install_data('etc/mpv-symbolic.svg', install_dir: join_paths(hicolor_dir, 'symbolic', 'apps'))
 
-    executable('mpv', sources, dependencies: dependencies, win_subsystem: 'windows,6.0',
-               include_directories: includedir, install: true)
+    mpv = executable('mpv', sources, dependencies: dependencies, win_subsystem: 'windows,6.0',
+                     include_directories: includedir, install: true)
+
+    # Check the protocols in the compiled binary and generate a .desktop file.
+    # Skip this if we're not on windows for simplicity.
+    if not win32
+        generate_desktop = find_program(join_paths(tools_directory, 'generate-desktop-file.py'))
+        custom_target('mpv.desktop',
+                      input: 'etc/mpv.desktop',
+                      output: 'mpv.desktop',
+                      depends: mpv,
+                      command: [generate_desktop, mpv.full_path(), '@INPUT@', '@OUTPUT@'],
```

Comment:
I think the full_path() prevents generation of the proper dependencies

Reply:
That's how I was originally doing it, but the problem is that `run_command` runs at configure time before the mpv binary is actually built.